.cirrus.yml

```
@@ -0,0 +1,201 @@
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+ubuntu_2004: &ubuntu_2004
+  container:
+    dockerfile: .cirrus/ubuntu2004/Dockerfile
+  env:
+    PATH: "${HOME}/.local/bin:${PATH}"
+    COV_COMPTYPE: gcc
+    COV_PLATFORM: linux64
+    GENERATOR: "Unix Makefiles"
+    BUILD_TOOL_OPTIONS: "-j 4"
+  pip_cache:
+    folder: ${HOME}/.cache/pip
+  conan_cache:
+    folder: ${HOME}/.conan/data
+  maven_cache:
+    folder: ${HOME}/.m2
+  bootstrap_script:
+    - pip3 install --user conan
+
+ubuntu_2004_gcc10: &ubuntu_2004_gcc10
+  <<: *ubuntu_2004
+  env:
+    CC: gcc-10
+
+ubuntu_2004_clang10: &ubuntu_2004_clang10
+  <<: *ubuntu_2004
+  env:
+    CC: clang-10
+
+freebsd12_clang8: &freebsd_12_clang8
+  freebsd_instance:
+    image_family: freebsd-12-1
+  env:
+    PATH: "${HOME}/.local/bin:${PATH}"
+    CC: clang
+    GENERATOR: "Unix Makefiles"
+    BUILD_TOOL_OPTIONS: "-j 4"
+  pip_cache:
+    folder: ${HOME}/.cache/pip
+  conan_cache:
+    folder: ${HOME}/.conan/data
+  maven_cache:
+    folder: ${HOME}/.m2
+  bootstrap_script:
+    - pkg update -f
+    - pkg install -y bash cmake maven openjdk11 perl5 py37-sqlite3 py37-pip
+    - pip install --user conan
+
+macos_1015_xcode11: &macos_1015_xcode11_5
+  osx_instance:
+    image: catalina-xcode-11.5
+  env:
+    PATH: "/usr/local/opt/openjdk/bin:${PATH}"
+    CC: clang
+    COV_COMPTYPE: clang
+    COV_PLATFORM: macOSX
+    GENERATOR: "Unix Makefiles"
+    BUILD_TOOL_OPTIONS: "-j 4"
+  pip_cache:
+    folder: ${HOME}/Library/Caches/pip
+  conan_cache:
+    folder: ${HOME}/.conan/data
+  maven_cache:
+    folder: ${HOME}/.m2
+  bootstrap_script:
+    - brew install bison cmake python3 maven
+    - python3 -m pip install --user conan
+    # Environment variables do not persist between scripts. Generate a proxy
+    # for Conan in /usr/local/bin/conan and forward the parameters.
+    - printf "#!/bin/bash\n$(python3 -m site --user-base)/bin/conan \"\$@\"\n" >> /usr/local/bin/conan
+    - chmod a+x /usr/local/bin/conan
+
+windows_2019_vs2019: &windows_2019_vs2019
+  windows_container:
+    dockerfile: .cirrus/windows2019/Dockerfile
+  env:
+    CIRRUS_SHELL: bash
+    COV_COMPTYPE: msvc
+    COV_PLATFORM: win64
+    BUILD_TOOL_OPTIONS: "-nologo -verbosity:minimal -maxcpucount -p:CL_MPCount=4"
+  pip_cache:
+    folder: ${LOCALAPPDATA}/pip/Cache
+  conan_cache:
+    folder: ${USERPROFILE}/.conan/data
+  maven_cache:
+    folder: ${USERPROFILE}/.m2
+  bootstrap_script:
+    - pip3 install --user conan
+
+env:
+  CYCLONEDDS_URI: '<CycloneDDS><DDSI2E><Internal><EnableExpensiveChecks>all</EnableExpensiveChecks></Internal></DDSI2E></CycloneDDS>'
```

That looks like an old one, on Travis we're using:

```suggestion
  CYCLONEDDS_URI: '<CycloneDDS><Domain><Internal><EnableExpensiveChecks>all</EnableExpensiveChecks><LivelinessMonitoring>true</LivelinessMonitoring></Internal><Tracing><Verbosity>config</Verbosity><OutputFile>stderr</OutputFile></Tracing></Domain></CycloneDDS>'
```